src/frontends/paddle/src/op/pool3d.cpp

```diff
@@ -0,0 +1,218 @@
+                        "pool3d kernel shape must be greater than 0");
+
+        // Note: this shape check is only valid when the spatial dim of input_shape
+        // is static.
+        if (input_shape[2].is_static() && input_shape[3].is_static() && input_shape[4].is_static()) {
+            uint64_t input_d = input_shape[input_rank - 3].get_length();
+            uint64_t input_h = input_shape[input_rank - 2].get_length();
+            uint64_t input_w = input_shape[input_rank - 1].get_length();
+            if ((input_d > 0) && (input_d + pad_begin[0] + pad_end[0] < kernel_d)) {
+                kernel_d = input_d + pad_begin[0] + pad_end[0];
+            }
+            if ((input_h > 0) && (input_h + pad_begin[1] + pad_end[1] < kernel_h)) {
+                kernel_h = input_h + pad_begin[1] + pad_end[1];
+            }
+            if ((input_w > 0) && (input_w + pad_begin[2] + pad_end[2] < kernel_w)) {
+                kernel_w = input_w + pad_begin[2] + pad_end[2];
+            }
+        }
+
+        if (pooling_type == "max") {
+            pool_outputs = std::make_shared<default_opset::MaxPool>(data,
+                                                                    ov::Strides(strides.begin(), strides.end()),
+                                                                    ov::Strides{1, 1, 1},
+                                                                    pad_begin,
+                                                                    pad_end,
+                                                                    ov::Shape{kernel_d, kernel_h, kernel_w},
+                                                                    rounding_type,
+                                                                    auto_pad)
+                               ->outputs();
+        } else {
+            bool exclude_pad = node.get_attribute<bool>("exclusive", false);
+            pool_outputs = std::make_shared<default_opset::AvgPool>(data,
+                                                                    ov::Strides(strides.begin(), strides.end()),
+                                                                    pad_begin,
+                                                                    pad_end,
+                                                                    ov::Shape{kernel_d, kernel_h, kernel_w},
+                                                                    exclude_pad,
+                                                                    rounding_type,
+                                                                    auto_pad)
+                               ->outputs();
+        }
+    }
+
+    if (data_format == "NDHWC") {
+        pool_outputs[0] = std::make_shared<default_opset::Transpose>(
+            pool_outputs[0],
+            std::make_shared<default_opset::Constant>(ov::element::i64, Shape{5}, std::vector<int64_t>{0, 2, 3, 4, 1}));
+    }
+
+    if (pool_outputs.size() == 1) {
```

Comment:
The number of outputs should be the same as output number of paddle op, not the number of openvino pool op.

Reply:
The tricky problem is that paddle maxpool op will have two outputs (Regardless of return_mask, [return_mask is implemented by python](https://github.com/PaddlePaddle/Paddle/blob/release/2.5/python/paddle/nn/functional/pooling.py#L1451))

But, when op is adapte && global_pool, openvino will use ReduceMax op. It cannot get the index. 

do you have any good advice?

Reply:
I check the python code, if want to return mask it will use 'max_pool3d_with_index'. So 'max_pool3d_with_index' has two outputs and 'pool3d' has one output. So in op_table.cpp, you can map 'max_pool3d_with_index' and 'pool3d' to the two different openvino frontend ops, they are different.

Reply:
I have updated it, use a new function to handle `max_pool_with_index`